sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java

```diff
@@ -236,7 +237,8 @@ public ServiceBusReceiverAsyncClient buildAsyncReceiverClient() {
         final TracerProvider tracerProvider = new TracerProvider(ServiceLoader.load(Tracer.class));
 
         return new ServiceBusReceiverAsyncClient(connectionProcessor.getFullyQualifiedNamespace(),
-            serviceBusResourceName, connectionProcessor, tracerProvider, messageSerializer, receiveMessageOptions);
+            serviceBusResourceName, MessagingEntityType.QUEUE, false, receiveMessageOptions,
+            connectionProcessor, tracerProvider, messageSerializer);
```

Comment:
`MessagingEntityType.QUEUE`  Is there a need to know the type ? We will not always know for example if user uses `connectionString(String connectionStringWithResourceName)` in Builder.

Reply:
Yes. We do when building the receiver itself. On management operations, they specify what the entity type is before sending the message (in the headers).